api/logging/v1/input_types.go

```diff
@@ -48,26 +44,33 @@ type Application struct {
 	// +optional
 	ContainerLimit *LimitSpec `json:"containerLimit,omitempty"`
 
-	// Containers is the spec of containers to include and exclude when collecting logs.
-	// This effectively is an 'AND' with the other fields
-	// of this input definition. The subfields of containers supports globs.  The includes
-	// defined here are combined with the included namespaces to form a set of namespace
-	// and container permutations.  The same logic is applied to the excluded containers and namespaces
+	// Includes is the set of namespaces and containers to include when collecting logs.
+	// Note: infrastructure namespaces are still excluded for "*" values and each must specifically included.
+	//
+	// +optional
+	Includes []IncludeExcludeSpec `json:"includes,omitempty"`
+
+	// Excludes is the set of namespaces and containers to ignore when collecting logs.
+	// Takes precedence over Includes option. The exclude patterns are applied after the attempt to glob everything in include.
 	//
 	// +optional
-	Containers *InclusionSpec `json:"containers,omitempty"`
+	Excludes []IncludeExcludeSpec `json:"excludes,omitempty"`
 }
 
-// InclusionSpec defines a set of similar resources for inclusion or exclusion
-type InclusionSpec struct {
+type IncludeExcludeSpec struct {
```

Comment:
IMO the better name for this struct is something like "NamespaceContainerSpec" given that is what it defines